app/src/main/java/org/sopt/pingle/presentation/ui/mygroup/MyGroupActivity.kt

```diff
@@ -1,12 +1,176 @@
+import android.view.View
+import androidx.activity.OnBackPressedCallback
+import androidx.activity.viewModels
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import org.sopt.pingle.R
 import org.sopt.pingle.databinding.ActivityMyGroupBinding
+import org.sopt.pingle.presentation.type.SnackbarType
+import org.sopt.pingle.presentation.ui.onboarding.OnBoardingActivity
 import org.sopt.pingle.util.base.BindingActivity
+import org.sopt.pingle.util.component.AllModalDialogFragment
+import org.sopt.pingle.util.component.PingleSnackbar
+import org.sopt.pingle.util.context.stringOf
 
+@AndroidEntryPoint
 class MyGroupActivity : BindingActivity<ActivityMyGroupBinding>(R.layout.activity_my_group) {
+
+    private val viewModel by viewModels<MyGroupViewModel>()
+    private lateinit var onBackPressed: OnBackPressedCallback
+    private lateinit var adapter: MyGroupAdapter
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        binding.myGroupviewModel = viewModel
+        binding.lifecycleOwner = this
+
+        viewModel.Logging()
+        initLayout()
+        addListeners()
+        collectData()
+        onBackPressedBtn()
+    }
+
+    override fun onDestroy() {
+        binding.rvMyGroupList.adapter = null
+        super.onDestroy()
+    }
+
+    private fun initLayout() {
+        checkMyGroupOwner()
+        viewModel.getGroupList()
+
+        with(binding) {
+            toolbar.text = getString(R.string.my_group_title)
+            // TODO chip 머지되면 text에 localStorage에 저장된 keyword 데바
+
+            adapter = MyGroupAdapter(this@MyGroupActivity::showChangeGroupModal)
+            rvMyGroupList.adapter = adapter
+            adapter.submitList(viewModel.filteredGroupList.value)
+        }
+    }
+
+    private fun addListeners() {
+        with(binding) {
+            toolbar.ivAllTopbarArrowWithTitleArrowLeft.setOnClickListener { finish() }
+            tvMyGroupMoveNewGroup.setOnClickListener { navigateToNewGroupInfo() }
+            ivMyGroupNowMenu.setOnClickListener { showMyGroupMenu() }
+            root.setOnClickListener { layoutMyGroupMenu.visibility = View.INVISIBLE }
+            layoutMyGroupMenuCopy.setOnClickListener { copyGroupCode() }
+            layoutMyGroupMenuShare.setOnClickListener { shareGroupCode() }
+        }
+    }
+
+    private fun collectData() {
+        viewModel.filteredGroupList.flowWithLifecycle(lifecycle).onEach { filteredList ->
+            adapter.submitList(filteredList)
+        }.launchIn(lifecycleScope)
+    }
+
+    private fun checkMyGroupOwner() {
+        with(binding) {
+            if (viewModel.getMyGroupIsOwner()) {
+                ivMyGroupNowOwner.visibility = View.VISIBLE
+            } else {
+                ivMyGroupNowOwner.visibility = View.INVISIBLE
+            }
```

Comment:
코드가 한줄인경우 대괄호를 빼는게 예쁜거같아요!
```suggestion
            if (viewModel.getMyGroupIsOwner()) ivMyGroupNowOwner.visibility = View.VISIBLE
            else ivMyGroupNowOwner.visibility = View.INVISIBLE
```

Reply:
이게ㅜㅜ 저도 그렇게 쓰고싶은데 컨트롤 엘이었나 린트였나 돌리면 일케 바뀌더라구요.. if~else문에 braces 없으면 안 좋다 어쩌구 하면서•• 참나!!

Reply:
저는 가독성 때문에 if, else 괄호 치는게 좋음 ㅋㅋ 개취인듯여 ~~